libethereum/BlockChain.cpp

```diff
@@ -122,6 +122,42 @@ class WriteBatchNoter: public ldb::WriteBatch::Handler
 };
 }
 
+namespace
+{
+
+class LastBlockHashes: public LastBlockHashesFace
+{
+public:
+	explicit LastBlockHashes(BlockChain const& _bc): m_bc(_bc) {}
+
+	h256s precedingHashes(h256 const& _mostRecentHash) const override
+	{
+		Guard l(m_lastHashesMutex);
+		if (m_lastHashes.empty() || m_lastHashes.back() != _mostRecentHash)
```

Comment:
Do you mean `front` instead of `back`?

In a following line I see `m_lastHashes[0] = _mostRecentHash`.